service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/common/Notification.java

```diff
@@ -58,13 +63,30 @@
 
     public static Notification fromKeyValueString(final String keyValueString, final String separatorPattern) {
 
-        Map<String, String> notificationValues = Splitter.onPattern(separatorPattern).withKeyValueSeparator("=").split(keyValueString);
+        final Map<String, String> notificationValues = Pattern.compile(separatorPattern)
+               .splitAsStream(keyValueString.trim())
+               .map(s -> s.split("="))
+               .collect(Collectors.toMap(a -> a[0], a -> a.length>1? a[1]: ""));
+
+        validateNotificationValues(notificationValues);
```

Comment:
Is this sth new?

Reply:
Actually, the recently used https://guava.dev/releases/18.0/api/docs/com/google/common/base/Splitter.html throws `IllegalArgumentException` if the specified sequence does not split into valid map entries, or if there are duplicate keys - Thats why i've added this validation.